server/jetstream.go

```diff
@@ -247,6 +248,25 @@ func (s *Server) getJetStream() *jetStream {
 	return js
 }
 
+func (a *Account) sendJSError(s *Server, event JSError) {
+	s.Warnf("  " + event.LogLine())
+
+	if event.ShouldRateLimit() && !errorEventLimit.Allow() {
+		return
+	}
+
+	advisory := event.Advisory()
+	advisory.Server = s.Name()
```

Comment:
Careful: s.Name() requires server lock. So make sure that where you call sendJSError, you are not under server lock or any lock that would cause lock inversion.
Alternatively, and since the server name is immutable, we could have a s.name() that just returns s.info.Name. If s.info is altered, we could if need be have a dedicated `name` field in the *Server structure and again have this immutable.

Reply:
Thanks will take a look - yeah the locking in server code is a bit hard to follow or to know what the model is, there seems no clear answer.  Many things - like name - that I think cant really change realistically at runtime is accessed under lock in other places.

If we decide to go with this pattern of code I'll have another look through these cases